modules/types/src/public.ts

```diff
@@ -12,12 +12,17 @@ type DepositParameters = {
   assetId?: Address; // if not provided, will default to 0x0 (Eth)
 };
 
-type DepositResponse = {
+type FreeBalanceResponse = {
   freeBalance: {
     [s: string]: BigNumber;
   };
 };
 
+type DepositResponse = {
+  txHash: string;
+  completed: () => Promise<FreeBalanceResponse>;
```

Comment:
Having the deposit promise resolve to the new free balance is cool.
It makes me wonder if we ought to change the name of that promise to indicate that it contains useful info, maybe `balanceUpdated` or `freeBalanceUpdateComplete`
idk I'm just being a stickler about the name of this promise lol it's not that important.

Reply:
Hmm, I'm inclined to agree but my issue is that it's a little misleading. It's not _just_ the balance being updated, the channel isn't usable until this process is complete, that's why `completed` makes more sense to me.